java/src/main/java/org/signal/libsignal/metadata/protocol/UnidentifiedSenderMessage.java

```diff
@@ -43,7 +40,7 @@ public UnidentifiedSenderMessage(byte[] serialized)
         throw new InvalidMetadataMessageException("Missing fields");
       }
 
-      this.ephemeral        = Curve.decodePoint(unidentifiedSenderMessage.getEphemeralPublic().toByteArray(), 0);
+      this.ephemeral        = new ECPublicKey(unidentifiedSenderMessage.getEphemeralPublic().toByteArray());
```

Comment:
Would you mind explaining this change?

Reply:
It'll make more sense after you look at my changes to libsignal-protocol-java, but Signal public keys have a key type prefix which is handled by that decodePoint function. I got rid of that because Signal only has one key type in practice, it seemed to complicate things unnecessarily and also forces a bunch of extra byte array copies.

By getting rid of that layer of indirection, I was able to make the interface that Curve.decodePoint produced (ECPublicKey) into a concrete type, which allows us to then just use its constructor rather than a factory method.

Reply:
This seems like a negative change IMO.  If Signal does introduce a new key type, we'd have to undo all of this to use it.  I do generally like getting rid of unnecessary indirection, but in the interest of maintainability and support for back-porting, I'd vote to keep things as close to the source library as possible.

Reply:
(I'll leave this to your judgement though)

Reply:
Keep in mind this is a hard fork, i don't plan to take future updates to this library. As far as I can tell, upstream is effectively frozen anyway (no commits in nearly a year, and new protocol dev work seems to be happening in Rust).

Reply:
Oh actually, libsignal-protocol-java which is where I made the root change hasn't had any commits since late 2019, so it's even more frozen.

Reply:
Okay, fair enough.